example/cripts/example1.cc

```diff
@@ -0,0 +1,242 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
```

Comment:
* As I review this, this example1.cc is the most important file in this patch. And I suspect this will be the case for others using this too who will go to this and hopefully other example Cript files. You have some great comments here, but **let's multiply them by 10.** At the start here, lets include an introductory comment explaining some of the philosophies of Cripts (I think you can copy and paste your description from this PR), and an explanation of what this example Cripts script does. I just skimmed through the script and it's not obvious to me yet what this does. Maybe it doesn't do anything specific other than highlight some features, which is fine for one example file.
* A question on terminology: are we calling these files "cripts", "scripts", "plugins", or something else? Maybe a "cript plugin"?
* Do you plan to add to the `docs/` for Cripts in a separate PR?

Reply:
1. There's some of this in the README that I added.
2. Cripts
3. Yes.